meteor/sofie-server/model/entities/test/entity-mock-factory.ts

```diff
@@ -0,0 +1,197 @@
+import { Segment, SegmentInterface } from '../segment'
+import { Part, PartInterface } from '../part'
+import { Piece, PieceInterface } from '../piece'
+import { PieceType } from '../../enums/piece-type'
+import { PieceLifespan } from '../../enums/piece-lifespan'
+import { Rundown } from '../rundown'
+import { anything, deepEqual, instance, mock, when } from 'ts-mockito'
+import { TransitionType } from '../../enums/transition-type'
+import { PartTimings } from '../../value-objects/part-timings'
+
+export class EntityMockFactory {
+	public static createActiveRundown(
```

Comment:
As this one is `mongo-rundown-repository.spec.ts` and my additions to `rundown-timeline-service.spec.ts` would not be able to make use of it.

If one were to try and insert multiple to the database of the entity instance gotten from this method, it would fail on the second, as the `id` would not be unique. Same seems to apply to all the other methods creating instances in this method.

The inability to chose whether the received rundown is active or not, also limits it.
This does seem to be chosen on a method name level, so either a new extremely similar method would need to exist, or this one should be refactored to allow giving in the desired active state via an argument.

i'd suggest renaming the method to `createRundown`, and adding the ability to supply the last 4 attributes that are currently hardcoded in the `when(..).thenReturn(...)`
The current values could then be the defaults, except for the `id`, where some randomly added text should be added after the base default id. I'd just suggest making the default for id be `rundownId + ${Math.random()}`
Something similar should also be done for the default id's in the other method that creates instances in this class.

I am fine with you not wanting to do it for this PR. 
In that case i would be applying the suggested changes when updating the tests to make use of this helper class.
Thought this was worth mentioning, so you have a chance to make the change yourself, including any that might be needed in your 100+ tests.

Reply:
It is allowed to refactor classes so they suit more needs ;)

In the spirit of YAGNI we only create the things that we actually need. When we discover that we need more, then we create it.

Reply:
Fair enought. I will be be refactoring it slightly when i get my hand on it, so it works for the repository test :)